habitat-lab/habitat/articulated_agent_controllers/humanoid_rearrange_controller.py

```diff
@@ -279,6 +296,80 @@ def calculate_walk_pose(
         self.obj_transform_base = obj_transform_base @ rot_offset
         self.joint_pose = joint_pose
 
+    def translate_and_rotate_with_gait(
+        self, forward_delta: float, rot_delta: float
```

Comment:
Because agent velocity is driven from config, we should consider having `forward_delta` a value between `-1.0` and `1.0`, and having this function multiply it with the agent's `lin_speed`.

If we do this, there would be a single source of truth to determine the humanoid agent's velocity.
This would eliminate any discrepancy caused by the implementer failing to read the right config field, and would ease agent config refactoring.

Reply:
This multiplication happens in the BaseVelocityAction, this way we have parity with how the robot moves around.
Here is where we calculate how much to move: https://github.com/facebookresearch/habitat-llm/blob/main_improve_planner/habitat_llm/agent/env/actions.py#L116


This is similar to the robot action, which computes delta and then transforms in here:
https://github.com/facebookresearch/habitat-lab/blob/main/habitat-lab/habitat/tasks/rearrange/actions/actions.py#L489